src/syringe.cr

```diff
@@ -9,18 +10,30 @@ module Syringe
 		macro finished
 			\{% for ctor in {{klass}}.methods.select { |m| m.name == "initialize" } %}
 				 def {{klass}}.new
+				    \{{ puts nodule = {{klass.id.split("::").reject { |k| k == klass.id.split("::").last}.join("::")}} }}
```

Comment:
@Bonemind  I not sure of a good way to make this a method to make this more readable. Also the `puts` was a workaround and I'm not sure of an elegant replacement to return nil but store the variable. I find macros hard to read in general so its possible I missed something obvious.

Reply:
Wow I realize I was being an idiot. still getting used to macros. Puts is fixed but would be nice to break this apart into helper methods to start or something. Would be nice to start building a more complicated framework with macro classes to improve readability.